app/controllers/analytics_controller.rb

```diff
@@ -33,99 +33,51 @@ def enroller_stats
-        symptomatic_patient_count_by_state_and_day[i][state] = count1
-        count2 = Assessment.joins(:patient).where('assessments.created_at::date = ?', d)
-                           .where('patients.monitored_address_state' => state).count
-        total_patient_count_by_state_and_day[i][state] = count2
-      end
-    end
-    date_map = {}
-    dates.each_with_index { |d, i| date_map[d] = i + 1 }
-    patient_count_by_day = Hash.new(0)
-    patients.each { |p| patient_count_by_day[date_map[p.created_at.to_date]] += 1 }
+    jurisdiction_analytics = current_user.jurisdiction.analytics
+    root_jurisdiction_analytics = current_user.jurisdiction.root.analytics
     patient_count_by_day_array = []
-    patient_count_by_day.each { |day, count| patient_count_by_day_array << { day: day, cases: count } }
-    patient_count_by_day_array.sort_by! { |count| count[:day] }
+    assessment_result_by_day_array = []
+    total_patient_count_by_state_and_day = []
+    symptomatic_patient_count_by_state_and_day = []
+    dates = (jurisdiction_analytics.pluck(:created_at).min.to_date..jurisdiction_analytics.pluck(:created_at).max.to_date).to_a
 
-    # Distribution by state for map
-    patient_count_by_state = Hash.new(0)
-    patients.each { |p| patient_count_by_state[p.monitored_address_state] += 1 }
+    dates.each do |date|
+      next if date.nil?
 
-    # Symptomatic or unsymptomatic per day
-    symptomatic_assessments_by_day = Hash.new(0)
-    asymptomatic_assessments_by_day = Hash.new(0)
-    Assessment.where(patient: patients).find_each do |a|
-      if a.symptomatic
-        symptomatic_assessments_by_day[a.created_at.to_date] += 1
-      else
-        asymptomatic_assessments_by_day[a.created_at.to_date] += 1
-      end
-    end
-    assessment_result_by_day_array = []
-    (symptomatic_assessments_by_day.keys | asymptomatic_assessments_by_day.keys).sort.each do |date|
+      # Get last saved analytic for each date
+      analytic = jurisdiction_analytics.where(created_at: date.beginning_of_day..date.end_of_day).last
+      root_analytic = root_jurisdiction_analytics.where(created_at: date.beginning_of_day..date.end_of_day).last
+      open_cases = !analytic&.open_cases_count.nil? ? analytic.open_cases_count : 0
+      patient_count_by_day_array << { day: date, cases: open_cases }
+      symp_count = !analytic&.symptomatic_monitorees_count.nil? ? analytic.symptomatic_monitorees_count : 0
       assessment_result_by_day_array << {
         'name' => date.to_s,
-        'Symptomatic Assessments' => symptomatic_assessments_by_day[date],
-        'Asymptomatic Assessments' => asymptomatic_assessments_by_day[date]
+        'Symptomatic Assessments' => symp_count,
+        'Asymptomatic Assessments' => open_cases - symp_count
       }
+      # Map analytics are pulled from the root (most likely USA) jurisdiction
+      sym_map = root_analytic&.monitoree_state_map.nil? ? (JSON.parse root_analytic.monitoree_state_map.gsub('=>', ':')) : {}
```

Comment:
I'm getting `NoMethodError (undefined method monitoree_state_map for nil:NilClass):` from this. Is the ternary swapped?

Reply:
This was a mistake introduced from fixing a rubocop error. Fixed now